tests/lib/c_lib/strerror/src/main.c

```diff
@@ -4,9 +4,7 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-#ifdef CONFIG_NEWLIB_LIBC
-#define _POSIX_C_SOURCE 200809
-#endif
+#define _POSIX_C_SOURCE 200809L
```

Comment:
sidenote: A benefit of #undef'ing  is that we don't need to worry about L or no L :) , so you won't get the conformance warning.

Reply:
I like leaving this alone -- it catches mistaken values used elsewhere in the system, which seems appropriate for a test.

Reply:
For the c library test that is indeed fair :)

Reply:
Similarly, if needed, it could be changed to something like this
```cpp
#if !defined(_POSIX_C_SOURCE) || _POSIX_C_SOURCE < [NEEDED_VER]
#undef _POSIX_C_SOURCE
#define _POSIX_C_SOURCE [NEEDED_VER]
#endif
```